### PR TITLE
Add service detection feature to node agent

### PR DESCRIPTION
**Pull request by nnDarshan**
This patch adds service detection capability to node-agent.
Node-agent will have a list of predefined services, which
it will look out for on the node in which it is running. It
will update etcd with the list of services available and or
running on the node. Also based on if the service is running
it will update the TAGS field in the node_context.

tendrl-bug-id: Tendrl/node_agent#136
Signed-off-by: nnDarshan <darshan.n.2024@gmail.com>

### CODE REVIEW

**Comment by nnDarshan** [comment]
@r0h4n @shtripat @nthomas-redhat Please review

**Comment by nnDarshan** [comment]
Unit Tests not added as UT will fail until https://github.com/Tendrl/commons/pull/101 is merged

File: tendrl/node_agent/manager/manager.py [hunk 1]
```diff
@@ -82,6 +84,18 @@ def _run(self):
         LOG.info("%s complete" % self.__class__.__name__)
 
 
+def update_node_context(manager, machine_id, tags=""):
```
Comment (nthomas-redhat) [inline tendrl/node_agent/manager/manager.py]: This should be in sync with https://github.com/Tendrl/node-agent/pull/134
one update should not overwrite the other

Comment (nnDarshan) [inline tendrl/node_agent/manager/manager.py]: This patch has been verified with, the patch you mentioned and it does not have any implication on that.

**Pull request merged**